crates/ty_python_semantic/src/types.rs

```diff
@@ -4450,6 +4489,11 @@ impl<'db> Type<'db> {
 
             Type::EnumLiteral(enum_literal) => enum_literal.enum_class_instance(db).bindings(db),
 
+            // This is a callable (it's a decorator)
+            Type::KnownInstance(known_instance @ KnownInstanceType::Deprecated(_)) => {
+                known_instance.instance_fallback(db).bindings(db)
+            }
+
             Type::PropertyInstance(_)
             | Type::KnownInstance(_)
```

Comment:
This fallback is valid for any known instance, we should implement it generally and not hardcode knowledge here about which specific ones are callable. For the ones that are not callable, we'll simply find that their class has no `__call__` in typeshed and continue to treat them as not-callable.

```suggestion
            Type::KnownInstance(known_instance) => {
                known_instance.instance_fallback(db).bindings(db)
            }

            Type::PropertyInstance(_)
```